src/main/kotlin/org/rust/ide/inspections/fixes/AddAssocTypeBindingsFix.kt

```diff
@@ -0,0 +1,82 @@
+ * found in the LICENSE file.
+ */
+
+package org.rust.ide.inspections.fixes
+
+import com.intellij.codeInspection.LocalQuickFixAndIntentionActionOnPsiElement
+import com.intellij.openapi.editor.Editor
+import com.intellij.openapi.project.Project
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import org.rust.ide.utils.template.buildAndRunTemplate
+import org.rust.lang.core.psi.*
+import org.rust.lang.core.psi.ext.elementType
+import org.rust.lang.core.psi.ext.getNextNonCommentSibling
+import org.rust.openapiext.createSmartPointer
+
+class AddAssocTypeBindingsFix(element: PsiElement) : LocalQuickFixAndIntentionActionOnPsiElement(element) {
+    override fun getText(): String = "Add missing associated types"
+    override fun getFamilyName() = text
+
+    override fun invoke(
+        project: Project,
+        file: PsiFile,
+        editor: Editor?,
+        startElement: PsiElement,
+        endElement: PsiElement
+    ) {
+        val element = startElement as? RsTraitRef ?: return
+        val arguments = element.path.typeArgumentList
+        val trait = element.path.reference?.resolve() as? RsTraitItem ?: return
+
+        val requiredAssocTypes = trait.associatedTypesTransitively
+            .filter { it.typeReference == null }
+            .associateBy { it.identifier.text }
+        val existingAssocTypes = arguments?.assocTypeBindingList
+            .orEmpty()
+            .associateBy { it.path.text }
+        val missingAssocTypes = requiredAssocTypes.filter { it.key !in existingAssocTypes }
+        if (missingAssocTypes.isEmpty()) return
+
+        val factory = RsPsiFactory(project)
+        val newAssocTypes = missingAssocTypes.toList().sortedBy { it.first }
+        val defaultType = "()"
+
+        val inserted = if (arguments != null) {
+            var anchor = with(arguments) {
+                assocTypeBindingList.lastOrNull() ?: typeReferenceList.lastOrNull() ?: lifetimeList.lastOrNull() ?: lt
+            }
+            val nextSibling = anchor.getNextNonCommentSibling()
+            val addCommaAfter = nextSibling?.isComma == true
+            if (addCommaAfter && nextSibling != null) {
+                anchor = nextSibling
+            }
+
+            for (type in newAssocTypes) {
+                if (anchor.elementType != RsElementTypes.LT && !anchor.isComma) {
+                    anchor = arguments.addAfter(factory.createComma(), anchor)
+                }
+                anchor = arguments.addAfter(factory.createAssocTypeBinding(type.first, defaultType), anchor)
+            }
+
+            if (addCommaAfter) {
+                arguments.addAfter(factory.createComma(), anchor)
+            }
+
+            arguments
+        } else {
+            val newArgumentList = factory.createTypeArgumentList(newAssocTypes.map { "${it.first}=$defaultType" })
+            element.path.addAfter(newArgumentList, element.path.identifier) as RsTypeArgumentList
+        }
```

Comment:
I refactored `AddGenericArguments` so its code can be reused. Please use it like this:
```kotlin
val arguments = argumentsExisting ?: path.addEmptyTypeArguments(factory)
val lastArgument = with(arguments) {
    (assocTypeBindingList + typeReferenceList + lifetimeList).maxByOrNull { it.startOffset } ?: lt
}
val missingTypes = missingTypesNames.map { factory.createAssocTypeBinding(it, defaultType) }
val addedArguments = arguments.addElements(missingTypes, lastArgument, factory)

editor?.buildAndRunTemplate(element, addedArguments.mapNotNull { it.typeReference?.createSmartPointer() })
```